src/AlertsManagement/AlertsManagement/SmartGroupCommands/GetAzureSmartGroup.cs

```diff
@@ -0,0 +1,111 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Linq;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Rest.Azure;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.AlertsManagement.OutputModels;
+using Microsoft.Azure.Management.AlertsManagement.Models;
+
+namespace Microsoft.Azure.Commands.AlertsManagement
+{
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SmartGroup", DefaultParameterSetName = SmartGroupsListByFilterParameterSet)]
+    [OutputType(typeof(PSSmartGroup))]
+    public class GetAzureSmartGroup : AlertsManagementBaseCmdlet
+    {
+        #region Parameter Set Names
+
+        private const string SmartGroupsListByFilterParameterSet = "SmartGroupsListByFilter";
+        private const string SmartGroupByIdParameterSet = "SmartGroupById";
+
+        #endregion
+
+        #region Parameters declarations
+        /// <summary>
+        /// SmartGroup Id
+        /// </summary>
+        [Parameter(Mandatory = true,
+                   ParameterSetName = SmartGroupByIdParameterSet,
+                   HelpMessage = "Unique Identifier of SmartGroup / ResourceId of SmartGroup.")]
+        [ValidateNotNullOrEmpty]
+        [Alias("ResourceId")]
+        public string SmartGroupId { get; set; }
```

Comment:
Same comment as abvove on Id - this shoudl break down the Resource Id into rgname + id or ResourceId, unless your API cannot get a single record usign rgname + (guid)id

Reply:
Alerts are proxy resources - they do not come under any particular resource group. Only Id is required for fetching alert. No resource group is associated with this resource.